src/raybcs/ReflectParticleBC.C

```diff
@@ -0,0 +1,54 @@
+//* This file is part of SALAMANDER: Software for Advanced Large-scale Analysis of MAgnetic
+// confinement for Numerical Design, Engineering & Research,
+//* A multiphysics application for modeling plasma facing components
+//* https://github.com/idaholab/salamander
+//* https://mooseframework.inl.gov/salamander
+//*
+//* SALAMANDER is powered by the MOOSE Framework
+//* https://www.mooseframework.inl.gov
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+//*
+//* Copyright 2025, Battelle Energy Alliance, LLC
+//* ALL RIGHTS RESERVED
+//*
+
+#include "ReflectParticleBC.h"
+#include "PICStudyBase.h"
+registerMooseObject("SalamanderApp", ReflectParticleBC);
+
+InputParameters
+ReflectParticleBC::validParams()
+{
+  auto params = ReflectRayBC::validParams();
+  params.addClassDescription(
+      "Reflective boundary condition for particles that ensures velocity data is consistent with "
+      "its direction after reflection.");
+  return params;
+}
+
+ReflectParticleBC::ReflectParticleBC(const InputParameters & params)
+  : ReflectRayBC(params),
+    _velocity_indicies(dynamic_cast<PICStudyBase *>(&_study)->getVelocityIndicies(false))
```

Comment:
```suggestion
    _velocity_indicies(getStudy<PICStudyBase>().getVelocityIndicies(false))
```

`dynamic_cast` will fail if the study isn't of the right type. `getStudy()` will give a useful error.